sympy/polys/galoistools.py

```diff
@@ -1,14 +1,18 @@
 """Dense univariate polynomials with coefficients in Galois fields. """
 
+import random
 from math import ceil as _ceil, sqrt as _sqrt, prod
 
 from sympy.core.random import uniform
-from sympy.external.gmpy import SYMPY_INTS
+from sympy.external.gmpy import SYMPY_INTS, invert
 from sympy.polys.polyconfig import query
 from sympy.polys.polyerrors import ExactQuotientFailed
 from sympy.polys.polyutils import _sort_factors
 
 
+rgen = random.Random()
```

Comment:
Should this move to sympy.core.random?

Reply:
I just imitated the way `random` is called elsewhere, so I will fix it if `core.random` should be used.

Reply:
This should be fixed. Just see how it is used elsewhere in the repo. In #25253 I made a comment where similar usage to this is employed -- so don't copy that ;-) I recall working on this in the crypto module, so check there as a starter.

Reply:
Thanks for the advice. Am I correct in using it this way?

Reply:
You imported the correct function but without allowing for a seed there is no way to control the behavior. Didn't the crypto functions that used random have a seed arg (maybe `_seed`)?